File: actividad hrencias/herncia/Automovil.py
from Vehiculo import Vehiculo

class Automovil(Vehiculo):
    def __init__(self, fecha_fabricacion, vin_chasis, vin_motor, marca, modelo, precio):
        super().__init__(fecha_fabricacion, vin_chasis, vin_motor)
        self.__marca = marca
        self.__modelo = modelo
        self.__precio = precio
    
    def obtener_marca(self):
        return self.__marca
    
    def obtener_modelo(self):
        return self.__modelo
    
    def obtener_precio(self):
        return self.__precio
    
    def mostrar_datos(self):
        print("Fecha de Fabricación:", self.obtener_fecha_fabricacion())
        print("VIN Chasis:", self.obtener_vin_chasis())
        print("VIN Motor:", self.obtener_vin_motor())
        print("Marca:", self.obtener_marca())
        print("Modelo:", self.obtener_modelo())
        print("Precio:", self.obtener_precio())

File: actividad hrencias/herencia/src/Automovil.java
public class Automovil extends Vehiculo {
    // Atributos específicos de la clase Automóvil
    private String marca;
    private String modelo;
    private double precio;
    
    
    public Automovil(String fechaFabricacion, String vinChasis, String vinMotor, 
                     String marca, String modelo, double precio) {
        super(fechaFabricacion, vinChasis, vinMotor);
        
        this.marca = marca;
        this.modelo = modelo;
        this.precio = precio;
    }
    
    public String obtenerMarca() {
        return this.marca;
    }
    
    public String obtenerModelo() {
        return this.modelo;
    }
    
    public double obtenerPrecio() {
        return this.precio;
    }
    
public void mostrarDatos() {
        
        System.out.println("INFORMACIÓN DEL AUTOMÓVIL ");
        System.out.println(" Fecha de Fabricación: " + obtenerFechaFabricacion());
        System.out.println(" VIN Chasis: " + obtenerVinChasis());
        System.out.println(" VIN Motor: " + obtenerVinMotor());
        System.out.println(" Marca: " + obtenerMarca());
        System.out.println("Molo: " + obtenerModelo());
        System.out.printf("Precio: $%,.2f%n", obtenerPrecio());
     
    }
}
    